src/blocks/gallery-masonry/save.js

```diff
@@ -76,7 +78,7 @@ const save = ( { attributes, className } ) => {
 							href = image.imgLink;
 						}
 
-						const img = <img src={ image.url } alt={ image.alt } data-id={ image.id } data-imglink={ image.imgLink } data-link={ image.link } className={ image.id ? `wp-image-${ image.id }` : null } />;
+						const img = <img src={ image.url } alt={ image.alt } data-id={ image.id } data-link={ image.link } data-imglink={ image.imgLink } data-index={ index } className={ image.id ? `wp-image-${ image.id }` : null } />;
```

Comment:
We shouldn't be utilizing the index of a loop here as there could be more than one gallery on a page.